ansible-ipi-install/roles/node-prep/tasks/02_bridge.yml

```diff
@@ -0,0 +1,75 @@
+---
+- name: install needed network manager libs
+  package:
+    name: "{{ item }}"
+    state: latest
+  with_items: "{{ nmcli_packages }}"
+
+- name: Delete {{ prov_nic }} due to modify nmcli bug
+  nmcli: 
+    conn_name: "{{ prov_nic }}"
+    type: ethernet
+    state: absent
+    
+- name: Create Bridge labeled provisioning
+  nmcli:
+    conn_name: provisioning
+    type: bridge
+    ifname: provisioning
+    autoconnect: yes
+    stp: off
+    ip4: 172.22.0.1/24
+    state: present
+
+- name: Create Bridge slave on {{ prov_nic }}
+  nmcli:
+    conn_name: "{{ prov_nic }}"
+    type: bridge-slave
+    hairpin: no
+    ip4_method: manual
+    ifname: "{{ prov_nic }}"
+    master: provisioning
+    autoconnect: yes
+    state: present
+    
+- name: Create Bridge labeled baremetal
+  nmcli:
+    conn_name: baremetal
+    type: bridge
+    ifname: baremetal
+    autoconnect: yes
+    stp: off
+    state: present
+
+- name: Create Bridge slave on {{ pub_nic }}
```

Comment:
does this configuration logic around the public_nic interrupt the ansible ssh connection? i remember having to make special provisions to ensure some configs were done atomically across multiple files/cmds over ssh because it would cause my ssh connection to drop for awhile.

Reply:
It actually doesn't interrupt the ansible ssh connection. Technically it is due to a bug in the nmcli.py file. Even though the nmcli module creates the bridges and slaves, when a file (ifcfg-*) already exists, and is modified, the module doesn't properly modify the file. Since it doesn't properly modify the file, I manually need to make edits to it. Since I make the edits to the file and the connection is active, I can then go ahead and use shell to ifdown and ifup the proper interfaces specific to the public nic. 

However, what I have is indeed a hack and in fact I included my own nmcli.py library file because there was bugs using the module against RHEL8 servers and it had no ip4_method available to it for manual connections. However, for our purposes and since I am supplying the module, it works as we'd want it to work (for now).

Reply:
> I can then go ahead and use shell to ifdown and ifup the proper interfaces specific to the public nic.

don't you at some point have to ifdown the public nic to pick up the changes though? I don't think I saw that happen.

Reply:
its in there
```
- name: Get DHCP for baremetal bridge
  shell: |
    /usr/sbin/ifdown {{ pub_nic }}; /usr/bin/pkill dhclient ; /usr/sbin/ifdown baremetal ; /usr/sbin/ifup {{ pub_nic }}; /usr/sbin/ifup baremetal
```